### ISSUE TITLE
Values from Typescript StringEnums and NumberEnums are not carried over to the generated code

### ISSUE THREAD

**Issue by ahodieb** [awslabs/jsii#251]
Typescript has String Enums and Numeric enums which associates a value to the Enum member 
[Details](https://www.typescriptlang.org/docs/handbook/enums.html)

**Example**
```typescript
export enum DynamoDBAttributeType {
    NUMBER = 'N',
    STRING = 'S',
    BINARY = 'B',
}

console.log(DynamoDBAttributeType.STRING)
>> S
```

The generated code doesn't provide any way to access those values.

**Generated Java**
```java
@javax.annotation.Generated(value = "jsii-pacmak/0.7.6 (build 04cb6bb)", date = "2018-09-30T14:18:02.519Z")
@software.amazon.jsii.Jsii(module = com.acme.hello.$Module.class, fqn = "hello-jsii.DynamoDBAttributeType")
public enum DynamoDBAttributeType {
    NUMBER,
    STRING,
    BINARY,
}
```

**Comment by rix0rrr** [comment]
That is true. Could you tell us what you need the enum values for?

**Comment by ahodieb** [comment]
I encountered a small problem caused by this issue while using the aws-cdk 

I'm using the java-cdk and i wanted to define a DynamoDB table with global secondary indexes, this functionality was not available in the Table construct (introduced by [this commit](https://github.com/awslabs/aws-cdk/commit/4980c972eaaebdc73b0d998f8f0b966e588673db) but not included in the cdk version that i am currently using) so i had to use the Table resource.

The table resource api is generated from the specs and expects the Attribute type of the key as a string 
```java
        TableResource table = new TableResource(this, "table", TableResourceProps.builder()
              ...
                .withAttributeDefinitions(asList(
                        TableResource.AttributeDefinitionProperty.builder()
                                .withAttributeName("id")
                                .withAttributeType(AttributeType.String.name().substring(0, 1))
                                .build(),
                        TableResource.AttributeDefinitionProperty.builder()
                                .withAttributeName("secondary_index")
                                .withAttributeType(AttributeType.String.name().substring(0, 1))
                                .build()))
                .build());
    }
```

The problem is `.withAttributeType()` expects a string value, and since there is now way to get the expect value from the `AttributeType` enum i had to do this `.withAttributeType(AttributeType.String.name().substring(0, 1))`

The problem is mainly with the generated cdk resources expecting a string instead of an enum in the first place (i'm going to add an issue for that there).

---
The problem with JSII is inconsistency of the generated code between different languages.
Values are available in typescript and javascript but not in java and dotnet. And i'm also curious how will it look in ruby/python/ho where there is no enum support in the language. 
I think at least this limitation should be stated somewhere in the docs.

**Comment by eladb** [comment]
I think the bug here is that `withAttributeType` is supposed to except an `AttributeType` and not a string. Looking into this.

**Comment by eladb** [comment]
This is fixed in the latest version of the CDK 0.15.1
